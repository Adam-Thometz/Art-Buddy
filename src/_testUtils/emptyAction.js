const EMPTY_ACTION = { type: "" }

export default EMPTY_ACTION