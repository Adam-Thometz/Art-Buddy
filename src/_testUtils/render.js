import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import GlobalContext from "_context/global";

import rootReducer from "_redux/rootReducer";

function withProvider({ children, store, initialRoutes }) {
  return (
    <Provider store={store}>
      <MemoryRouter initialEntries={initialRoutes}>
        <GlobalContext>{children}</GlobalContext>
      </MemoryRouter>
    </Provider>
  );
}

function renderWithProvider(component, options) {
  const store = options && options.store ? options.store : rootReducer;
  const initialRoutes = options && options.initialRoutes ? options.initialRoutes : ["/"];
  return render(component, {
    wrapper: (args) => withProvider({ ...args, store, initialRoutes }),
    ...options,
  });
}

export * from "@testing-library/react";

export { renderWithProvider as render };
