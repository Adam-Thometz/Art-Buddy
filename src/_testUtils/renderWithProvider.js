import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;