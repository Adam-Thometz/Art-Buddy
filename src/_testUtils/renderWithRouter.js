import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import '@testing-library/jest-dom';

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return (
    <MemoryRouter initialEntries={initialRoutes}>
      {children}
    </MemoryRouter>
  );
};

function renderWithRouter(component, options) {
  const initialRoutes = (options && options.initialRoutes)
    ? options.initialRoutes
    : ['/'];
  return render(component, { wrapper: (args) => 
    MemoryRouterWithInitialRoutes({
      ...args,
      initialRoutes
    }), ...options
  });
};

export default renderWithRouter;