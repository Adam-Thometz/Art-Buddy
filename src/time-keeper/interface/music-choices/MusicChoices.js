import "./MusicChoices.css";

import MusicOption from "time-keeper/interface/music-choices/music-option/MusicOption";

import timeBlocks from "_data/time-keeper/timeBlocks";
import { EXCITED, RELAXED } from "_assets/time-keeper/timeKeeper.assets";

const MusicChoices = ({ mood }) => {
  const moodSrc = mood === "excited" ? EXCITED : RELAXED;
  return (
    <div className="MusicChoices">
      <img src={moodSrc} alt="" />
      {timeBlocks.map((block) => (
        <MusicOption mood={mood} seconds={block.seconds} music={block[mood]} />
      ))}
    </div>
  );
};

export default MusicChoices;
