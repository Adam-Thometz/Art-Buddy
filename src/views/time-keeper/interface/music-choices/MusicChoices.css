.MusicChoices {
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-size: 40px;
  padding: 25px;
}
