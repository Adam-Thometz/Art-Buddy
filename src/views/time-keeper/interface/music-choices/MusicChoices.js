import "./MusicChoices.css";

import MusicOption from "./music-option/MusicOption";

import timeBlocks from "data/time-keeper/timeBlocks";
import { EXCITED, RELAXED } from "assets/time-keeper/timeKeeper.assets";

const MusicChoices = ({ mood }) => {
  const moodSrc = mood === "excited" ? EXCITED : RELAXED;
  return (
    <div className="MusicChoices">
      <img src={moodSrc} alt="" />
      {timeBlocks.map((block) => (
        <MusicOption mood={mood} seconds={block.seconds} music={block[mood]} />
      ))}
    </div>
  );
};

export default MusicChoices;
