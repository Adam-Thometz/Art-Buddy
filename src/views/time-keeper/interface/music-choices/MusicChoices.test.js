import MusicChoices from "./MusicChoices";

import { render } from "_testUtils/render";

describe("MusicChoices component", () => {
  const SAMPLE_MOOD = "excited";
  it("renders without crashing", () => {
    render(<MusicChoices mood={SAMPLE_MOOD} />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<MusicChoices mood={SAMPLE_MOOD} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
