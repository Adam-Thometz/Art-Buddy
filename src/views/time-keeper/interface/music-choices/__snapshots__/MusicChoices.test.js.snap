// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicChoices component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MusicChoices"
  >
    <img
      alt=""
      src="file"
    />
    <div
      class="MusicOption excited"
    >
      15 SEC
    </div>
    <div
      class="MusicOption excited"
    >
      30 SEC
    </div>
    <div
      class="MusicOption excited"
    >
      1 MIN
    </div>
    <div
      class="MusicOption excited"
    >
      3 MIN
    </div>
    <div
      class="MusicOption excited"
    >
      5 MIN
    </div>
  </div>
</DocumentFragment>
`;
